lib/datadog/core/configuration/settings.rb

```diff
@@ -212,11 +212,13 @@ def initialize(*_)
             # This was used prior to the GA of the new CPU Profiling 2.0 profiler. The CPU Profiling 2.0 profiler does not
             # use or need this setting and thus it doesn't do anything.
             option :max_events do |o|
-              o.after_set do
-                Datadog.logger.warn(
-                  'The profiling.advanced.max_events setting has been deprecated for removal and no ' \
-                  'longer does anything. Please remove it from your Datadog.configure block.'
-                )
+              o.after_set do |_, _, precedence|
+                unless precedence == Datadog::Core::Configuration::Option::Precedence::DEFAULT
```

Comment:
Without this, a simple `get` on this option (which we do because of the profiler settings dump) will trigger this codepath: 

https://github.com/DataDog/dd-trace-rb/blob/28dc18783c6dd6e6740c180e71a190346e75d9c0/lib/datadog/core/configuration/option.rb#L117

Which will eventually trigger `after_set` even for unset/default values:

https://github.com/DataDog/dd-trace-rb/blob/28dc18783c6dd6e6740c180e71a190346e75d9c0/lib/datadog/core/configuration/option.rb#L299

An alternative could be to just check for a nil value as well. But given that we'll be introducing more and more remote config support which means yet another type of precedence, having that info available in the `after_set` block sounds interesting.

Reply:
Yeah... I'm not particularly excited about adding more features to the options system, but I do agree that this seems like it would be relevant for remote configuration and the only alternative I could think of:

```ruby
# doing something like this in the collector itself
(v.class.options.keys - [:force_enable_new_profiler, :legacy_transport_enabled, :force_enable_legacy_profiler, :allocation_counting_enabled]).each { |key| v.get_option(key) }
```

...is super ugly as well.

My suggestion is to keep the approach you have now, but please don't forget to add test coverage for the new feature in `option_spec` ;)